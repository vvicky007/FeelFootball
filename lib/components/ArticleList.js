import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

ArticleList.propTypes = {
    
};

function ArticleList(props) {
    return (
        <div>
           {Object.values(props.articles).map((article)=>(
           
               <Article key={article.id} article={article} action={props.articleActions}/>
           ))}
        </div>
    );
}

export default ArticleList;