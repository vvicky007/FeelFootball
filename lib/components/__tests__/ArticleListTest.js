import React from 'react'
import ArticleList from '../ArticleList'
import renderer from 'react-test-renderer'
describe('Article List',()=>{
    const testProps = {
        articles:{
            a:{id:'a'},
            b:{id:'b'}
        },
        articleActions:{
            lookup:jest.fn(()=>({}))
        }
    }
    it('renderscorrectly',()=>{
        const tree = renderer.create(<ArticleList {...testProps}/>).toJSON()
        
        expect(tree.children.length).toBe(2)
        expect(tree).toMatchSnapshot();
    })
   
})