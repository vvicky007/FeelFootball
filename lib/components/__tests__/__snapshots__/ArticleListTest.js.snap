// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article List renderscorrectly 1`] = `
<div>
  <div>
    <div />
    <div />
    <div>
      <a>
         
      </a>
    </div>
    <div />
  </div>
  <div>
    <div />
    <div />
    <div>
      <a>
         
      </a>
    </div>
    <div />
  </div>
</div>
`;
